pyrit/prompt_converter/azure_speech_text_to_audio_converter.py

```
@@ -0,0 +1,113 @@
+        Saves the file to the results/audio folder
+
+        Raises:
+            ValueError: Any issues in validation or execution.
+        """
+        if prompt == "":
+            raise ValueError("Prompt was empty. Please provide valid input prompt.")
+        try:
+            speech_config = speechsdk.SpeechConfig(subscription=self.azure_speech_key, region=self.azure_speech_region)
+            speech_config.speech_synthesis_language = self.synthesis_language
+            speech_config.speech_synthesis_voice_name = self.synthesis_voice_name
+            if output_format == "mp3":
+                speech_config.set_speech_synthesis_output_format(
+                    speechsdk.SpeechSynthesisOutputFormat.Audio16Khz32KBitRateMonoMp3
+                )
+            if not self.filename:
+                self.filename = f"{uuid.uuid4()}.wav"
+            if not os.path.isdir(self.output_dir):
+                os.mkdir(self.output_dir)
+            file_name = os.path.join(self.output_dir, self.filename)
+            file_config = speechsdk.audio.AudioOutputConfig(filename=file_name)
+            speech_synthesizer = speechsdk.SpeechSynthesizer(speech_config=speech_config, audio_config=file_config)
+            result = speech_synthesizer.speak_text_async(prompt).get()
+            if result.reason == speechsdk.ResultReason.SynthesizingAudioCompleted:
+                logger.info(
+                    "Speech synthesized for text [{}], and the audio was saved to [{}]".format(prompt, file_name)
+                )
+            elif result.reason == speechsdk.ResultReason.Canceled:
+                cancellation_details = result.cancellation_details
+                logger.info("Speech synthesis canceled: {}".format(cancellation_details.reason))
+                if cancellation_details.reason == speechsdk.CancellationReason.Error:
+                    logger.error("Error details: {}".format(cancellation_details.error_details))
```

These two cases look like errors to me. We can't proceed from this point onward since the converter failed, or did I misunderstand by any chance? If I read this right we should raise an exception here.

And the log message command for cancelation should probably be `logger.error`, too.

---

This is from the Microsoft example code, they consider the cancel as an operation and not an error.

---

For their use cases it may be that way. But for us it's an exception since we can't continue.

---

added exception